dagshub/common/config.py

```diff
@@ -61,3 +61,12 @@
         f"Number of download threads was set to {download_threads}. "
         f"We recommend lowering the value if you get met with rate limits"
     )
+
+
+def set_host(new_host: str):
+    _parsed_host = urlparse(new_host)
+    _hostname = _parsed_host.hostname
+    _host = _parsed_host.geturl().rstrip("/")
+
+    global hostname, host, parsed_host
+    hostname, host, parsed_host = _hostname, _host, _parsed_host
```

Comment:
Shouldn't this logic be reused somehow in the piece of code where you read DAGSHUB_CLIENT_HOST ?

Reply:
Honestly not a huge fan of how it looks like with the reuse, but sure.